src/arrayMethodSort.js

```diff
@@ -4,8 +4,32 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    if (compareFunction === undefined) {
+      for (let i = 0; i < this.length - 1; i++) {
+        if (this[i] + '' > this[i + 1] + '') {
+          const currentIndex = this[i];
+
+          this[i] = this[i + 1];
+          this[i + 1] = currentIndex;
+
+          i = -1;
+        }
+      }
+    } else {
+      for (let i = 0; i < this.length - 1; i++) {
+        if (compareFunction(this[i], this[i + 1]) > 0) {
+          const currentIndex = this[i];
+
+          this[i] = this[i + 1];
+          this[i + 1] = currentIndex;
+
+          i = -1;
+        }
+      }
+    }
+
+    return this;
```

Comment:
Returning 'this' is correct for chaining, but ensure that the sorting algorithm works correctly before returning.